crates/bevy_tasks/src/priority_executor.rs

```diff
@@ -0,0 +1,70 @@
+use std::future::Future;
+use std::sync::atomic::AtomicUsize;
+
+use async_executor::{Executor, Task};
+use event_listener::Event;
+use futures_lite::FutureExt;
+
+/// Task priority.
+#[repr(usize)]
+#[derive(Debug, Clone, Copy)]
+pub enum Priority {
+    FinishWithinFrame = 0,
+    AcrossFrame = 1,
+    IO = 2,
+}
+
+#[derive(Debug)]
+pub struct PriorityExecutor<'a> {
+    ex: [Executor<'a>; 3],
+    event: Event,
+    worker: AtomicUsize,
```

Comment:
Just as a note, this `worker` member is never used in the implementation.
If its not needed, we should remove it.